### PR TITLE
Upgrade to VS 2017

### PR DESCRIPTION
**Pull request by natemcmaster**
Remaining issues:

 - tests fail on net46
   
   ```
   System.MissingMethodException : Method not found: 'System.Net.Http.HttpClient Microsoft.AspNetCore.TestHost.TestServer.CreateClient()'.
   ```

 - Consider using [MSBuild typescript targets](https://www.typescriptlang.org/docs/handbook/compiler-options-in-msbuild.html). 

 - Use version BuildBundlerMinifier in samples instead of dotnet-bundle when https://github.com/madskristensen/BundlerMinifier/issues/230 is fixed

cref https://github.com/aspnet/Universe/issues/469

cc @anurse

### CODE REVIEW

**Comment by anurse** [comment]
I'm going to disable `net46` tests for now. The tests are failing due to https://github.com/Microsoft/vstest/issues/428 as far as I can tell.

**Comment by anurse** [comment]
I'd also suggest we do the TypeScript MSBuildification later and focus on just getting this merged in. The conversion will be the most disruptive change, not many people are working on the TS client right now. It's really just @moozzyk  ;).

**Comment by moozzyk** [comment]
Yes. I will look into TS client.

File: src/Microsoft.Extensions.WebSockets.Internal/Microsoft.Extensions.WebSockets.Internal.csproj [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <Import Project="..\..\build\common.props" />
+
+  <PropertyGroup>
+    <Description>Low-allocation Push-oriented WebSockets based on Channels</Description>
+    <VersionPrefix>0.1.0</VersionPrefix>
+    <TargetFramework>netstandard1.3</TargetFramework>
+    <NoWarn>$(NoWarn);CS1591</NoWarn>    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```
Comment (moozzyk) [inline src/Microsoft.Extensions.WebSockets.Internal/Microsoft.Extensions.WebSockets.Internal.csproj]: M&Ms

File: test/Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest/Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest.csproj [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <Import Project="..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp1.1</TargetFramework>  </PropertyGroup>
```
Comment (moozzyk) [inline test/Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest/Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest.csproj]: ditto

Comment (natemcmaster) [inline test/Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest/Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest.csproj]: Simple way to coalesce line endings? I've hit this a few times before while migrating ...not sure why.

File: src/Microsoft.Extensions.WebSockets.Internal/Microsoft.Extensions.WebSockets.Internal.csproj (hunk 1 above)
Comment (anurse) [inline src/Microsoft.Extensions.WebSockets.Internal/Microsoft.Extensions.WebSockets.Internal.csproj]: mmmmmm

**Comment by natemcmaster** [comment]
Added workaround for microsoft/vstest#428. 

Next issue. Microsoft.AspNetCore.SignalR.Tests on net46 appears to hang indefinitely.

**Comment by BrennanConroy** [comment]
@anurse Was looking into it

**Comment by natemcmaster** [comment]
Removing "wip" label. I got a full build to pass on command-line 🎆

**Comment by anurse** [comment]
Rebased. There was a random macOS failure in the Autobahn tests but I think it's a flaky test error. I'm going to file a bug to look in to stabilizing that. One option might be to look at migrating WebSockets back to its own repo so the Autobahn suite doesn't make the whole SignalR repo flaky

**Comment by pranavkm** [comment]
This is tangentially related to the work here, but in the CoreCLR (mirroring) build we hit long path issues with the config file:

```
C:\BuildAgent\work\29495d0a71cf35ac\Universe\SignalR\test\Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest>dir obj\Debug\net46\*.config

Directory of C:\BuildAgent\work\29495d0a71cf35ac\Universe\SignalR\test\Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest\obj\Debug\net46

02/03/2017  05:47 AM               427 Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest.csproj.Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest.dll.config

C:\BuildAgent\work\29495d0a71cf35ac\Universe\SignalR\test\Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest>dir obj\Debug\net46\obj\Debug\net46\Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest.csproj.Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest.dll.config
The filename or extension is too long.
```

This manifests in the config file not being copied to the output dir - from the build log:

```
Target "GenerateBindingRedirectsUpdateAppConfig" skipped, due to false condition; 
    ('$(AutoGenerateBindingRedirects)' == 'true' and '$(GenerateBindingRedirectsOutputType)' == 'true' and Exists('$(_GenerateBindingRedirectsIntermediateAppConfig)')) 
was evaluated as 
    ('true' == 'true' and 'true' == 'true' and Exists('obj\Debug\net46\Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest.csproj.Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest.dll.config')).`
```

**Comment by natemcmaster** [comment]
We should open an issue on the SDK for this. Looks like they double-nested the location: `obj\Debug\net46\obj\Debug\net46\Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest.csproj.Microsoft.AspNetCore.WebSockets.Internal.ConformanceTest.dll.config`

Simple fix: shorten the name of the test projects.

**Pull request merged**